redskyctl/internal/commands/authorize_cluster/generator.go

```diff
@@ -76,15 +76,31 @@ func NewGeneratorCommand(o *GeneratorOptions) *cobra.Command {
 		RunE: commander.WithContextE(o.generate),
 	}
 
+	// Provide a more meaningful default client name if possible
+	if o.ClientName == "" {
```

Comment:
is it worth moving this into `generate()` so we can `s/context.TODO()/cmd.Context()/`?

Reply:
I put it here so we could include the answer as the default value for the CLI flag (i.e. it shows up in `--help`). The main thing attached to the `cmd.Context()` is the UA strings for HTTP requests which we don't need here. I used `TODO()` instead of `Background()` because the usage better fits the description.

That being said, when I moved it, it still uses `cfg.Kubectl` which is going to be problematic because we haven't hit `preRun` yet (nor will we for `--help`) so the `cfg` hasn't been loaded yet. I'll change it to exec `kubectl` directly in this case.